modules/Microsoft.Purview/accounts/deploy.bicep

```diff
@@ -0,0 +1,394 @@
+@description('Optional. Azure location where the Purview Account will be created.')
+param location string = resourceGroup().location
+
+@description('Required. Name of the Purview Account.')
+param name string
+
+@description('Optional. Resource Tags.')
+param tags object = {}
+
+@description('Optional. The ID(s) to assign to the resource.')
+param userAssignedIdentities object = {}
+
+@description('Required. The Managed Resource Group Name.')
+param managedResourceGroupName string
+
+@description('Optional. Enable or disable resource provider inbound network traffic from public clients. default is Disabled.')
+@allowed([
+  'Enabled'
+  'Disabled'
+  'NotSpecified'
+])
+param publicNetworkAccess string = 'Disabled'
+
+@description('Optional. Specifies the number of days that logs will be kept for; a value of 0 will retain data indefinitely.')
+@minValue(0)
+@maxValue(365)
+param diagnosticLogsRetentionInDays int = 365
+
+@description('Optional. Resource ID of the diagnostic storage account. For security reasons, it is recommended to set diagnostic settings to send data to either storage account, log analytics workspace or event hub.')
+param diagnosticStorageAccountId string = ''
+
+@description('Optional. Resource ID of the diagnostic log analytics workspace. For security reasons, it is recommended to set diagnostic settings to send data to either storage account, log analytics workspace or event hub.')
+param diagnosticWorkspaceId string = ''
+
+@description('Optional. Resource ID of the diagnostic event hub authorization rule for the Event Hubs namespace in which the event hub should be created or streamed to.')
+param diagnosticEventHubAuthorizationRuleId string = ''
+
+@description('Optional. Name of the diagnostic event hub within the namespace to which logs are streamed. Without this, an event hub is created for each log category. For security reasons, it is recommended to set diagnostic settings to send data to either storage account, log analytics workspace or event hub.')
+param diagnosticEventHubName string = ''
+
+@description('Optional. Array of role assignment objects that contain the \'roleDefinitionIdOrName\' and \'principalId\' to define RBAC role assignments on this resource. In the roleDefinitionIdOrName attribute, you can provide either the display name of the role definition, or its fully qualified ID in the following format: \'/providers/Microsoft.Authorization/roleDefinitions/c2f4ef07-c644-48eb-af81-4b1b4947fb11\'.')
+param roleAssignments array = []
+
+@description('Conditional. Existing Subnet Resource ID to assign to the Private Endpoint. Required if Private Endpoints are required.')
+param subnetId string = ''
+
+@description('Conditional. Name of the Purview Account Private Endpoint. Required if the Purview account Private Endpoint is required.')
```

Comment:
Regarding Portal and Account PE: Need to align to the private endpoint implementation pattern in CARML.
- Similar to your approach, use the Private endpoint module in CARML.
- Remove the parameters: endpoint name, nic name and endpoint IP.
- Pass this information as an array as it can support more than 1 PE type. See [Example](https://github.com/Azure/ResourceModules/blob/646a947a7c42ec462aa223066f6dc860dadf60df/modules/Microsoft.Storage/storageAccounts/deploy.bicep#L340). You can also multiple PE services in that same array without having to use multiple parameters for each type. See [test implementation](https://github.com/Azure/ResourceModules/blob/646a947a7c42ec462aa223066f6dc860dadf60df/modules/Microsoft.Automation/automationAccounts/.test/common/deploy.test.bicep#L93) that passes in multiple service types (account and portal)

Regarding the Ingestion endpoints (blob and queue): Similar to the above, but the service resource Id in the private endpoints needs to be defined inside the module as an output of the managed storage account resource ID: 

Example: 
``` bicep
serviceResourceId: account.managedResources.storageAccount
```

![image](https://user-images.githubusercontent.com/28486158/216723003-f5b86220-2428-4955-aef8-18b1e5cd83b1.png)

![image](https://user-images.githubusercontent.com/28486158/216723125-f1f7af26-c441-4025-a0f3-c3e92b2cffc3.png)

Reply:
updated